lib/interface_quda.cpp

```diff
@@ -1842,6 +1872,457 @@ void dslashQuda(void *h_out, void *h_in, QudaInvertParam *inv_param, QudaParity
   popVerbosity();
 }
 
+void shiftQuda(void *h_out, void *h_in, int dir, int sym, QudaInvertParam inv_param)
+{
+  profileCovDev.TPSTART(QUDA_PROFILE_TOTAL);
+  profileCovDev.TPSTART(QUDA_PROFILE_INIT);
+
+  const auto &gauge = *gaugePrecise; //(inv_param->dslash_type != QUDA_ASQTAD_DSLASH) ? *gaugePrecise : *gaugeFatPrecise;
+
+  inv_param.solution_type = QUDA_MAT_SOLUTION;
+  inv_param.dirac_order = QUDA_DIRAC_ORDER;
+
+  if (!gaugePrecise)
+    errorQuda("Gauge field not allocated");
+
+  pushVerbosity(inv_param.verbosity);
+  if (getVerbosity() >= QUDA_DEBUG_VERBOSE) printQudaInvertParam(&inv_param);
+
+  ColorSpinorParam cpuParam(h_in, inv_param, gauge.X(), false, inv_param.input_location);
+  ColorSpinorField *in_h = ColorSpinorField::Create(cpuParam);
```

Comment:
Allocate on stack instead of heap

Reply:
fixed in 4fa8f65983905f955fa1dabad997b25e447653df